website/project/views/contributor.py

```diff
@@ -822,6 +823,18 @@ def claim_user_form(auth, **kwargs):
             if provider:
                 redirect_url = web_url_for('auth_login', next=provider.landing_url, _absolute=True)
             else:
+                node = AbstractNode.load(pid)
+                preprint = Preprint.load(pid)
+                if node:
+                    node_source_tags = node.all_tags.filter(name__icontains='source:', system=True)
+                    if node_source_tags.exists():
+                        for tag in node_source_tags:
+                            claimed_tag, created = Tag.all_tags.get_or_create(name=NODE_SOURCE_TAG_CLAIMED_TAG_RELATION[tag.name], system=True)
```

Comment:
Will this need to be updated for registries when registries have providers?

Reply:
The registry tags will need to be updated when registry provider becomes a thing.